### PR TITLE
Gracefully Handle unbound viewholders during on onChildDraw of EpoxyModelTouchCallback

### PR DESCRIPTION
**Pull request by podarsmarty**
- Converted EpoxyTouchHelperCallback to kotlin
- Updated EpoxyModelTouchCallback to handle unbound viewholders during onChildDraw gracefully

### CODE REVIEW

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyTouchHelperCallback.kt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+     * @see canDropOver
+     */
+    protected open fun canDropOver(
+        recyclerView: RecyclerView,
+        current: EpoxyViewHolder,
+        target: EpoxyViewHolder
+    ) =
+        super.canDropOver(recyclerView, current, target)
+
+    override fun getSwipeThreshold(viewHolder: RecyclerView.ViewHolder) =
```
Comment (elihart) [inline epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyTouchHelperCallback.kt:65]: can you add explicit return types for all of these functions? best to do that for public library interfaces

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelTouchCallback.java [hunk 2]
```diff
@@ -206,26 +206,30 @@ protected void onChildDraw(Canvas c, RecyclerView recyclerView, EpoxyViewHolder
       float dX, float dY, int actionState, boolean isCurrentlyActive) {
     super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive);
 
-    EpoxyModel<?> model = viewHolder.getModel();
```
Comment (elihart) [inline epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelTouchCallback.java:209]: can you limit the try /catch to just `viewHolder.getModel()` to reduce scope?

**Review by elihart — APPROVED** [review]

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyTouchHelperCallback.kt (hunk 1 above)
Comment (podarsmarty) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyTouchHelperCallback.kt:65]: sure, done

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelTouchCallback.java (hunk 2 above)
Comment (podarsmarty) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelTouchCallback.java:209]: done

**Pull request merged**